xml/security_pcidss.xml

```diff
@@ -281,75 +281,71 @@
-        the <literal>FW_CUSTOMRULES</literal> line.
-       </para>
-       <para>
-        To add an outbound rule, add an appropriate
-        <command>iptables</command> command line inside the
-        <literal>fw_custom_after_chain_creation()</literal> function. This
-        function hook is executed during the firewall setup and allows any
-        customized <systemitem>iptables</systemitem> rule.
+        Make the SSH daemon only reachable on the administration
+        interface and not on the general network card if possible.
+        Define the source addresses that a service allows traffic from.
        </para>
        <para>
         For example, to allow only outbound DNS requests over the
         interface <systemitem>eth0</systemitem> to server
-        <systemitem class="ipaddress">10.0.0.4</systemitem>, use:
+        <systemitem class="ipaddress">10.0.0.1</systemitem>, use:
        </para>
-       <screen>&prompt.sudo;iptables -A OUTPUT -d 10.0.0.4/32 -o eth0 -p udp -m udp --dport 53 -j ACCEPT</screen>
+<screen>&prompt.root;<command>firewall-cmd</command> --permanent --direct --add-rule ipv4 filter OUTPUT 5 \
+    -d 10.0.0.1/32 -p udp -m udp --dport 53 -j ACCEPT
+&prompt.root;<command>firewall-cmd</command> --reload</screen>
        <para>
-        Also see the <literal>deny all OUTPUT</literal> rule described
-        in <xref linkend="vle.pcidss.deny-all"/>.
+        To block all other outbound traffic, see <xref linkend=
+        "vle.pcidss.deny-all" />.
        </para>
       </listitem>
      </varlistentry>
      <varlistentry xml:id="vle.pcidss.deny-all">
       <term>
-       1.2.1.c Examine firewall and router configurations to verify that all
-       other inbound and outbound traffic is specifically denied, for example
-       by using an explicit <quote>deny all</quote> or an implicit
-       <quote>deny after allow</quote> statement.
+       1.2.1.c Verify that all other inbound and outbound traffic is
+       specifically denied.
       </term>
       <listitem>
        <para>
-        The <quote>deny all</quote> rules of other inbound and outbound
-        traffic can easily be achieved with iptables. The
-        <literal>INPUT</literal> and <literal>FORWARD</literal> table
-        policies are directly set by the &susefirewall; script, so all
-        unwanted traffic is dropped. Forwarding is usually completely
-        disabled by a kernel parameter and should not be enabled for endpoint
-        servers.
+        Deny all outbound and inbound traffic for which no exceptions
+        are defined as stated in the previous section. Forwarding is
+        usually completely disabled by a kernel parameter and should
+        not be enabled for endpoint servers.
+       </para>
+       <para>
+        To block all inbound traffic, set your Ethernet interface to
+        the zone <literal>block</literal>. Also allow already
+        established connections to persist:
+       </para>
```

Comment:
The firewalld default configuration already blocks all not specifically allowed incoming traffic. It has also the connection tracking rules. So there is no need to add them manually. Only the traffic for services that are explicitly enabled inside firewalld can come through.

By default all outgoing traffic is simply allowed. To change that we need to enable connection tracking on the OUTPUT chain and then drop all packet at the end.

The connection tracking rule need to be at the top of the chain. Therefore it get the priority of 0.
`firewall-cmd --permanent --direct --add-rule ipv4 filter OUTPUT 0 -m conntrack --ctstate RELATED,ESTABLISHED -j ACCEPT`

The DROP rule for the OUTPUT chain should be at the end of the chain and therefore gets a high priority number.
`firewall-cmd --permanent --direct --add-rule ipv4 filter OUTPUT 99 -j DROP`

The `iptables` filter rules will look like this:
```
-A OUTPUT -j OUTPUT_direct
-A OUTPUT_direct -m conntrack --ctstate RELATED,ESTABLISHED -j ACCEPT
-A OUTPUT_direct -j DROP
```
And inside the firewalld XML configuration the rules look like this:
```
<?xml version="1.0" encoding="utf-8"?>
<direct>
  <rule ipv="ipv4" table="filter" chain="OUTPUT" priority="0">-m conntrack --ctstate RELATED,ESTABLISHED -j ACCEPT</rule>
  <rule ipv="ipv4" table="filter" chain="OUTPUT" priority="99">-j DROP</rule>
</direct>
```

Reply:
Maybe it is a good idea to show the firewalld configuration as well, so that the administrator can double check them after the command line was executed.